src/wallet/wallet.cpp

```diff
@@ -4854,6 +4854,21 @@ void CWallet::ListLockedCoins(std::vector<COutPoint>& vOutpts) const
     }
 }
 
+void CWallet::ListProTxCoins(int height, std::vector<COutPoint>& vOutpts)
+{
+    auto mnList = deterministicMNManager->GetListForBlock(chainActive[height]);
+
+    AssertLockHeld(cs_wallet);
```

Comment:
i would make 4861-4869 to a method so i can be reused for void CWallet::ListProTxCoins(std::vector<COutPoint>& vOutpts), they have the same code